pkg/gui/controllers/tags_controller.go

```diff
@@ -190,6 +193,16 @@ func (self *TagsController) push(tag *models.Tag) error {
 		FindSuggestionsFunc: self.c.Helpers().Suggestions.GetRemoteSuggestionsFunc(),
 		HandleConfirm: func(response string) error {
 			return self.c.WithWaitingStatus(self.c.Tr.PushingTagStatus, func(task gocui.Task) error {
+				self.c.State().SetRefOperation(tag.FullRefName(), types.RefOperationPushing, context.TAGS_CONTEXT_KEY)
+				defer func() {
+					self.c.State().ClearRefOperation(tag.FullRefName(), context.TAGS_CONTEXT_KEY)
+					// Render again to remove the inline status:
```

Comment:
shouldn't this be done from within `stopRenderingInlineStatus`?

Reply:
No, this would come too early and cause the ugly flicker again (not for this command, but for the other ones). Most other commands need to refresh first and only redraw again at the end.

Reply:
TL;DR makes sense, happy to leave as-is

I see: what differentiates this from other situations is that in other cases we'll do a refresh at the end of the callback passed to WithItemOperation because we expect the model to change as a result of the action. With tags, we don't keep track of which remotes a tag has been pushed to, so there's no need to refresh. But in this case we can't piggyback on the fact that a refresh causes a re-render because we're not refreshing. That's why we need to do an explicit render.

This actually shines a light on the fact that we _could_ be storing remote tags but we just aren't. Once we support that, we'll add a refresh here and then we won't need the call to HandleRender.

I was wondering whether we should re-render anyway within WithItemOperation but it's an extra render we don't need to do so I'm happy to leave this as is.